Bricks/strings/util.h

```diff
@@ -25,33 +25,56 @@ SOFTWARE.
 #ifndef BRICKS_STRINGS_UTIL_H
 #define BRICKS_STRINGS_UTIL_H
 
-#include <cstddef>
-#include <type_traits>
-#include <cstring>
-#include <cctype>
 #include <algorithm>
-#include <iterator>
+#include <cstring>
+#include <sstream>
+#include <string>
 
-// These are small headers, and #include "bricks/strings/util.h" should be enough.
-#include "printf.h"
-#include "fixed_size_serializer.h"
-#include "join.h"
-#include "split.h"
-#include "distance.h"
-#include "chunk.h"
+#include "is_string_type.h"
+
+#include "../template/decay.h"
 
 namespace bricks {
 namespace strings {
 
+template <typename DECAYED_T>
+struct ToStringImpl {
+  template <typename T>
+  static std::string ToString(T&& something) {
+    static_assert(std::is_same<bricks::decay<T>, DECAYED_T>::value, "");
+    return std::to_string(std::forward<T>(something));
```

Comment:
`std::forward` is overkill here - `std::to_string` accepts only a number of numeric types.